common/lib/dependabot/pull_request_creator/bitbucket_server.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require "dependabot/clients/bitbucket_server"
+
+module Dependabot
+  class PullRequestCreator
+    class BitbucketServer
+      attr_reader :source, :branch_name, :base_commit, :credentials,
+                  :files, :commit_message, :pr_description, :pr_name,
+                  :reviewers
+
+      def initialize(source:, branch_name:, base_commit:, credentials:,
+                     files:, commit_message:, pr_description:, pr_name:,
+                     reviewers: nil)
+        @source         = source
+        @branch_name    = branch_name
+        @base_commit    = base_commit
+        @credentials    = credentials
+        @files          = files
+        @commit_message = commit_message
+        @pr_description = pr_description
+        @pr_name        = pr_name
+        @reviewers      = reviewers
+      end
+
+      def create
+        return if branch_exists? && pull_request_exists?
```

Comment:
My earlier comment about not having an updater is incomplete. This logic isn't quite right.

```ruby
      def pull_request
        @pull_request ||= client_for_source.
                          fetch_pull_request_for_branch(branch_name)
      end

      def pull_request_exists?
        pull_request.nil? ? false : true
      end
```

`pull_request_exists?` will return true if the branch exists and there is a PR, but `@base_commit` might not exist in this branch.

Looking at https://github.com/dependabot/dependabot-core/blob/main/common/lib/dependabot/pull_request_creator/github.rb#L39, `#create` calls `#create_annotated_pull_request` which calls `#create_or_update_branch(commit)` which calls `#update_branch(commit)` which adds a commit to the branch.

Looks like we can do something similar here.